placement/algo/sharded_helper.go

```diff
@@ -230,10 +260,15 @@ func (ph *placementHelper) targetLoadForInstance(id string) int {
 }
 
 func (ph *placementHelper) moveOneShard(from, to placement.Instance) bool {
-	for _, s := range from.Shards().All() {
-		if s.State() != shard.Leaving && ph.moveShard(s, from, to) {
-			return true
-		}
+	// Prefer to move a Unknown shard first, then Initializing, then Available.
```

Comment:
Can you elaborate on the rationale behind this heuristic?

Reply:
Modified the comment